framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1869,7 +1869,7 @@ protected void checkTypecastSafety(TypeCastTree typeCastTree) {
      * @param exprType annotated type of the casted expression
      * @return true if the type cast is safe, false otherwise
      */
-    private boolean isTypeCastSafe(AnnotatedTypeMirror castType, AnnotatedTypeMirror exprType) {
+    protected boolean isTypeCastSafe(AnnotatedTypeMirror castType, AnnotatedTypeMirror exprType) {
```

Comment:
If you're going to make this method protected, why not do the modifications to the types above in an override of this method, instead.

That would have two advantages:
* you would not need to override `checkTypecastSafety`, and so you wouldn't need to duplicate its logic
* the implementation would be simpler, since you would already have the two `AnnotatedTypeMirror`s. You could then just make the changes and then call super with the modified types

Reply:
Thanks. I pushed a new commit with the changes.